src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,100 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+interface MovieProps {
+  onAdd: (movie: Movie) => void;
+}
+export const NewMovie: React.FC<MovieProps> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const areAllFieldsFilled = (title !== '')
+    && (imgUrl !== '')
```

Comment:
You don't have to compare empty string to empty string in this case, empty string is falsy itself
```suggestion
    && (imgUrl)
```

Reply:
done